subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ProviderCodecs.kt

```diff
@@ -162,14 +162,12 @@ class BuildServiceProviderCodec(
 
     override suspend fun WriteContext.encode(value: BuildServiceProvider<*, *>) {
         encodePreservingSharedIdentityOf(value) {
-            val buildIdentifier = value.buildIdentifier
-            write(buildIdentifier)
-            writeString(value.name)
-            writeClass(value.implementationType)
-            write(value.parameters)
-            writeInt(
-                buildServiceRegistryOf(buildIdentifier).forService(value).maxUsages
-            )
+            val serviceDetails = value.serviceDetails
+            write(serviceDetails.getBuildIdentifier())
```

Comment:
💄 Prefer property syntax over method invocation (`serviceDetails.buildIdentifier`, etc)

Reply:
So... I didn't see any problems in the IDE, however `compileKotlin` seems to get confused by Java fields and their corresponding getters:

```
> Task :configuration-cache:compileKotlin FAILED
e: /Users/rafael/sources/releng-gradle/subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ProviderCodecs.kt: (166, 34): Cannot access 'buildIdentifier': it is private in 'BuildServiceDetails'
e: /Users/rafael/sources/releng-gradle/subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ProviderCodecs.kt: (167, 40): Cannot access 'name': it is private in 'BuildServiceDetails'
e: /Users/rafael/sources/releng-gradle/subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ProviderCodecs.kt: (168, 39): Cannot access 'implementationType': it is private in 'BuildServiceDetails'
e: /Users/rafael/sources/releng-gradle/subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ProviderCodecs.kt: (169, 34): Cannot access 'parameters': it is private in 'BuildServiceDetails'
e: /Users/rafael/sources/releng-gradle/subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/ProviderCodecs.kt: (170, 37): Cannot access 'maxUsages': it is private in 'BuildServiceDetails'
```

That surprised me, but I worked it around by using the getter names. Any ideas?

Reply:
Hm, declaring the local variable type explicitly works around that:

```
val serviceDetails: BuildServiceDetails<*, *> = value.serviceDetails
```